File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03945/p03945_num6_parsed.py
# Program:

def min_new_stones(S):
    n = len(S) # Number of stones on the board
    black_count = 0 # Count of black stones
    white_count = 0 # Count of white stones

    for stone in S:
        if stone == 'B':
            black_count += 1
        else:
            white_count += 1
    
    if black_count == 0 or white_count == 0:
        return 0
    
    if S[0] == 'B' and S[-1] == 'W':
        return min(black_count, white_count)
    elif S[0] == 'W' and S[-1] == 'B':
        return max(black_count, white_count)
    else:
        return max(black_count, white_count) - 1

# Getting the input
S = input()

# Output the minimum number of new stones that Jiro needs to place
print(min_new_stones(S))

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03945/p03945_num6_parsed.cpp
#include <iostream>
#include <string>
using namespace std;

int getMinNewStones(string s) {
    int countB = 0; // count of black stones
    int countW = 0; // count of white stones
    
    for (char c : s) {
        if (c == 'B') {
            countB++;
        } else {
            countW++;
        }
    }
    
    // If all stones are already of the same color
    if (countB == 0 || countW == 0) {
        return 0;
    }
    
    // Minimum number of new stones that Jiro needs to place
    return min(countB, countW);
}

int main() {
    string s;
    cin >> s;
    
    int minNewStones = getMinNewStones(s);
    cout << minNewStones << endl;
    
    return 0;
}